### PR TITLE
Improving the no token error during setup

### PR DESCRIPTION
**Pull request by orta**
///////////////////⚡///////////////////

Danger is sleeping, so I don't have too much time, but this is the start of improving the no token setup error messaging. Will finish tomorrow.

### CODE REVIEW

**Comment by k0nserv** [comment]
What about optional variables that you might want to set? Do we wanna raise any info about those too?

**Comment by orta** [comment]
It wouldn't fail here if they were missing

**Comment by k0nserv** [comment]
Nope, but it could be a good opportunity to surface them as well?

File: lib/danger/danger_core/environment_manager.rb [hunk 1]
```diff
@@ -67,5 +67,41 @@ def self.danger_head_branch
...
+    def raise_error_for_no_request_source
+      title = ""
+      subtitle = ""
+      repo = ci_source.repo_url
+      case repo
+      when repo =~ /github/
+        title = "For your GitHub repo, you need to expose:" + RequestSources::GitHub.env_vars.join(", ").yellow
+      when repo =~ /gitlab/
+        title = "For your GitLab repo, you need to expose:" + RequestSources::GitLab.env_vars.join(", ").yellow
+
```
Comment (JuanitoFatas) [inline lib/danger/danger_core/environment_manager.rb]: ✂️

File: lib/danger/danger_core/environment_manager.rb [hunk 2]
```diff
@@ -67,5 +67,41 @@ def self.danger_head_branch
...
+        " - #{klass}, #{klass.env_vars.join(', ').yellow}"
+        end
+        title = "For Danger to run on this project, you need to expose some of following the ENV vars\n#{available.join('\n')}"
+      end
+
+      if ci_source.class == Danger::Travis
+        subtitle = "If you have an open source project, you should ensure 'Display value in build log' enabled, so that PRs from forks work."
+      end
+
+      repo_url.
```
Comment (JuanitoFatas) [inline lib/danger/danger_core/environment_manager.rb]: This looks weird 😕❓

**Comment by JuanitoFatas** [comment]
Do we also want to add `.env_vars` to the [base class](https://github.com/danger/danger/blob/master/lib/danger/request_sources/request_source.rb) and raise an not implemented error? 😊

**Comment by orta** [comment]
allllright, this should be good for review now. Danger is out all day, so I get some other Danger time.

**Comment by orta** [comment]
All feedback agreed with and fixed

**Comment by JuanitoFatas** [comment]
🛳⛵🚢⛴🛥🚤

**Comment by orta** [comment]
works for me

**Pull request merged**